### PR TITLE
ATO-211: Generate the correct identifier

### PR DESCRIPTION
**Pull request by ethanmills**
## What?

Use just the host of the internal sector uri to generate the internal pairwise identifier used in the authentication user info response

### CODE REVIEW

File: auth-external-api/src/test/java/uk/gov/di/authentication/external/services/UserInfoServiceTest.java [hunk 1]
```diff
@@ -144,7 +147,7 @@ private static Stream<Arguments> provideTestData() {
     private static AccessTokenStore getMockAccessTokenStore(List<String> claims) {
         var accessTokenStore = mock(AccessTokenStore.class);
         when(accessTokenStore.getSubjectID()).thenReturn(TEST_SUBJECT.getValue());
-        when(accessTokenStore.getSectorIdentifier()).thenReturn(TEST_RP_SECTOR_URI);
+        when(accessTokenStore.getSectorIdentifier()).thenReturn(TEST_RP_SECTOR_HOST);
```
Comment (ethanmills) [inline auth-external-api/src/test/java/uk/gov/di/authentication/external/services/UserInfoServiceTest.java:150]: This reflects the real behaviour. This value comes from the access token store, which comes from the auth code store, which comes from the auth frontend request. That is set to the host here:https://github.com/govuk-one-login/authentication-frontend/blob/main/src/components/auth-code/auth-code-service.ts#L44

I've also verified this by checking the access token store dynamo in sandpit

File: integration-tests/src/test/java/uk/gov/di/authentication/api/AuthExternalApiUserInfoIntegrationTest.java [hunk 2]
```diff
@@ -200,7 +200,7 @@ void shouldReturn401ForAccessTokenThatIsPastItsTtl() {
     private UserProfile addTokenToDynamoAndCreateAssociatedUser(
             String accessToken, List<String> claims, boolean isNewAccount) {
         accessTokenStoreExtension.addAccessTokenStore(
-                accessToken, TEST_SUBJECT.getValue(), claims, isNewAccount, RP_SECTOR_ID_URI);
+                accessToken, TEST_SUBJECT.getValue(), claims, isNewAccount, RP_SECTOR_ID_HOST);
```
Comment (ethanmills) [inline integration-tests/src/test/java/uk/gov/di/authentication/api/AuthExternalApiUserInfoIntegrationTest.java:203]: See comment on other test file

**Review by billrdunn — DISMISSED** [review]

File: auth-external-api/src/main/java/uk/gov/di/authentication/external/services/UserInfoService.java [hunk 3]
```diff
@@ -32,13 +32,13 @@ public UserInfo populateUserInfo(AccessTokenStore accessTokenInfo) {
         String internalSubjectId = accessTokenInfo.getSubjectID();
         var userProfile = authenticationService.getUserProfileFromSubject(internalSubjectId);
 
-        String internalRpPairwiseId =
-                ClientSubjectHelper.calculatePairwiseIdentifier(
-                        internalSubjectId,
+        Subject internalRpPairwiseId =
```
Comment (CarlyG55) [inline auth-external-api/src/main/java/uk/gov/di/authentication/external/services/UserInfoService.java:35]: While we're here, could you rename this to remove `Rp`? I think it makes it quite unclear

Comment (CarlyG55) [reply auth-external-api/src/main/java/uk/gov/di/authentication/external/services/UserInfoService.java:35]: (otherwise happy for this to go in!

**Review by dbes-gds — APPROVED** [review]

**Review by CarlyG55 — APPROVED** [review]

**Pull request merged**